Formula/o/openjdk@8.rb

```diff
@@ -123,25 +133,28 @@ def install
         --with-toolchain-type=clang
         --with-zlib=system
       ]
-
-      # Work around SDK issues with JavaVM framework.
-      if MacOS.version <= :catalina
-        sdk_path = MacOS::CLT.sdk_path(MacOS.version)
-        ENV["SDKPATH"] = ENV["SDKROOT"] = sdk_path
-        javavm_framework_path = sdk_path/"System/Library/Frameworks/JavaVM.framework/Frameworks"
-        args += %W[
-          --with-extra-cflags=-F#{javavm_framework_path}
-          --with-extra-cxxflags=-F#{javavm_framework_path}
-        ]
-        ldflags << "-F#{javavm_framework_path}"
-      # Fix "'JavaNativeFoundation/JavaNativeFoundation.h' file not found" issue on MacOS Sonoma.
-      elsif MacOS.version == :sonoma
-        javavm_framework_path = "/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks"
-        args += %W[
-          --with-extra-cflags=-F#{javavm_framework_path}
-          --with-extra-cxxflags=-F#{javavm_framework_path}
-        ]
-        ldflags << "-F#{javavm_framework_path}"
+      # Work around Xcode 16 bug: https://bugs.openjdk.org/browse/JDK-8340341
+      DevelopmentTools.clang_build_version == 1600 &&
+        ENV.append_to_cflags("-mllvm -enable-constraint-elimination=0")
+      # Pre-build JNF (JavaNativeFoundation.framework) in macOS Sonoma or newer
+      # to fix the issue of its necessary headers not being found.
+      if MacOS.version >= :sonoma
+        ENV["SDKPATH"] = ENV["SDKROOT"] = MacOS::CLT.sdk_path(MacOS.version)
+        resource("JavaNativeFoundation").stage do
+          cd "apple/JavaNativeFoundation" do
+            xcodebuild "-arch", Hardware::CPU.arch,
+                       "OTHER_CFLAGS=-Wno-strict-prototypes",
+                       "-project", "JavaNativeFoundation.xcodeproj"
+            my_jnf_path = buildpath/"build/Frameworks"
+            my_jnf_path.install "build/Release/JavaNativeFoundation.framework"
```

Comment:
Given the `tbd` file is still there in the SDK and it's just the headers that are missing - have you tried just `-isystem` the headers directory here rather than building the whole framework?

Reply:
I haven’t tried it yet. It sounds like a feasible approach, but I first need to find a source that can be downloaded to these headers.